### PR TITLE
Theme config does not take effect

### PR DESCRIPTION
**Pull request by lcduong**
> Fixes [#137 Theme config does not take effect](https://github.com/fossasia/eventyay-video/issues/137)
> 
> #### Short description of what this resolves:
> Theme config does not take effect
> 
> #### Changes proposed in this pull request:
> * Update theme.config.colors with colors data from API response
> * Change color default for theme
> 
> #### Checklist
> * [x]  I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
> * [x]  My branch is up-to-date with the Upstream `development` branch.
> * [ ]  The acceptance, integration, unit tests and linter pass locally with my changes
> * [ ]  I have added tests that prove my fix is effective or that my feature works
> * [ ]  I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by lcduong** [comment]
image


Note: for upload logo file, [urls] [media] must be configured in config file
e.g:
[urls]
media=http://localhost:8443/media/

since /media is expose by server app, so need to config nginx for expose via webapp.

File: server/venueless/zoom/views.py [hunk 1]
```diff
@@ -53,7 +54,8 @@ def get_closest_zoom_lang(world):
 class ZoomViewMixin:
     @cached_property
     def world(self):
-        w = get_object_or_404(World, domain=self.request.headers["Host"])
+        world_domain = re.sub(r":\d+$", "", self.request.headers["Host"])
```
Comment (hongquan) [inline server/venueless/zoom/views.py:57]: Please use Django's built-in `request.get_host()`.

File: server/venueless/api/views.py [hunk 2]
```diff
@@ -82,6 +83,21 @@ def patch(self, request, **kwargs):
...
+
+    def get(self, request, **kwargs):
+        """
+        Retrieve theme config of a world
+        @param request: request obj
+        @param kwargs: world_id
+        @return: theme data of a world
+        """
+        world = get_object_or_404(World, id=kwargs["world_id"])
+        return Response(WorldSerializer(world).data['config']['theme'])
```
Comment (hongquan) [inline server/venueless/api/views.py:98]: Please handle `KeyError`.

**Review by hongquan — CHANGES_REQUESTED** [review]

**Comment by hongquan** [comment]
I think it is not safe to assume that the API response is always in the shape that the frontend expects (like missing some fields, or different data type).
It is better to go through a "validation step" (using https://docs.superstructjs.org/ or https://zod.dev/). This process will show us (developers) how the API is unmatched, so that we can fix frontend, or backend code easier. But I'm not sure if we should require our frontend developers to apply this practice (I feel it is a bit advance).

In the future, we can migrate our frontend to Vue 3 + TypeScript, this "validation step" will be mandatory, because not only it help developers to find where data is wrong, but also provide type information (TypeScript) for IDE to produce good autocomplete.

File: server/venueless/api/views.py [hunk 3]
```diff
@@ -82,6 +83,25 @@ def patch(self, request, **kwargs):
...
+        """
+        Retrieve theme config of a world
+        @param request: request obj
+        @param kwargs: world_id
+        @return: theme data of a world
+        """
+        try:
+            world = get_object_or_404(World, id=kwargs["world_id"])
+            return Response(WorldSerializer(world).data['config']['theme'])
+        except KeyError:
```
Comment (hongquan) [inline server/venueless/api/views.py:100]: If `KeyError` happens, it means that us, the website owner, did not prepare enough data, or misconfigured the website. We need to do `logger.error` to inform our team.

For end user, we can throw error 503 to tell that this error is our side, temporary and can work again in the future.

**Review by hongquan — CHANGES_REQUESTED** [review]

File: server/venueless/api/views.py [hunk 4]
```diff
@@ -82,6 +86,26 @@ def patch(self, request, **kwargs):
...
+            logger.error(f"error happened when trying to get theme data of world: " + kwargs["world_id"])
```
Comment (hongquan) [inline server/venueless/api/views.py:104]: Incorrect use of logger. Check guide in https://github.com/fossasia/eventyay-talk/pull/130

**Review by hongquan — CHANGES_REQUESTED** [review]

**Review by hongquan — APPROVED** [review]

**Pull request merged**